airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py

```diff
@@ -0,0 +1,199 @@
+        yield from response.json()
+
+class IncrementalWoocommerceStream(WoocommerceStream, ABC):
+    # Getting page size as 'limit' from parrent class
+    @property
+    def limit(self):
+        return super().limit
+
+    # Setting the check point interval to the limit of the records output
+    state_checkpoint_interval = limit
+    # Setting the default cursor field for all streams
+    cursor_field = "date_modified"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+        # If there is a next page token then we should only send pagination-related parameters.
+        if not next_page_token:
+            params["orderby"] = self.order_field
+            params["order"] = "asc"
+            if stream_state:
+                params["after"] = stream_state.get(self.cursor_field)
+        return params
+
+    # Parse the stream_slice with respect to stream_state for Incremental refresh
+    # cases where we slice the stream, the endpoints for those classes don't accept any other filtering,
+    # but they provide us with the updated_at field in most cases, so we used that as incremental filtering during the order slicing.
+    def filter_records_newer_than_state(self, stream_state: Mapping[str, Any] = None, records_slice: Mapping[str, Any] = None) -> Iterable:
+        # Getting records >= state
+        if stream_state:
+            for record in records_slice:
+                if record[self.cursor_field] >= stream_state.get(self.cursor_field):
+                    yield record
+        else:
+            yield from records_slice
+
+class NonFilteredStream(IncrementalWoocommerceStream):
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+
+        if not next_page_token and stream_state:
+            del params["after"]
+        return params
+
+    def read_records(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+        ) -> Iterable[Mapping[str, Any]]:
+
+            slice = super().read_records(sync_mode = SyncMode.full_refresh, stream_slice = stream_slice,stream_state = stream_state)
+            yield from self.filter_records_newer_than_state(stream_state=stream_state, records_slice=slice)
+
+
+class Coupons(IncrementalWoocommerceStream):
+    data_field = "coupons"
+    order_field = "date"
```

Comment:
You have it specified in base class, redefine `order_field` only if necessary. You can remove it from this class.